pkgs/applications/misc/shavee/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, pkgs, rustPlatform, fetchFromGitHub, pkg-config, libiconv, openssl, linux-pam }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "shavee";
+  version = "v0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "ashuio";
+    repo = "shavee";
+    rev = "${version}";
+    sha256 = "sha256-41wJ3QBZdmCl7v/6JetXhzH2zF7tsKYMKZY1cKhByX8=";
+  };
+
+  cargoSha256 = "sha256-j0wRfRG6TU9LBhhCWf+cBxNICO1s9pyOQn0tm2Lrr0Y=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+      libiconv
+      linux-pam
+      openssl
+      openssl.dev
+
+  ];
+
+  buildPhase = ''
```

Comment:
why not use the default phase?

Reply:
the default phase does not create the output path.

Reply:
not sure what you mean.

```nix
{ lib
, rustPlatform
, fetchFromGitHub
, pkg-config
, openssl
, zlib
, stdenv
, pam
}:

rustPlatform.buildRustPackage rec {
  pname = "shavee";
  version = "0.5.1";

  src = fetchFromGitHub {
    owner = "ashuio";
    repo = "shavee";
    rev = "v${version}";
    hash = "sha256-41wJ3QBZdmCl7v/6JetXhzH2zF7tsKYMKZY1cKhByX8=";
  };

  cargoHash = "sha256-iNGn5KknSNgazFSu6Nur7AkKVb6qKMxuFwTdCz5djWU=";

  nativeBuildInputs = [
    pkg-config
  ];

  buildInputs = [
    openssl
    zlib
    pam
  ];

  # these tests require network access
  checkFlags = [
    "--skip=filehash::tests::remote_file_hash"
    "--skip=filehash::tests::get_filehash_unit_test"
  ];

  meta = with lib; {
    description = "shavee is a Program to automatically decrypt and mount ZFS datasets using Yubikey HMAC as 2FA or any File on USB/SFTP/HTTPS drive with support for PAM to auto mount home directories";
    homepage = "https://github.com/ashuio/shavee";
    license = licenses.mit;
    platforms = platforms.linux;
    maintainers = with maintainers; [ ];
  };
}
```
appears to build fine for me.

Reply:
aah, I see. I introduced `buildPhase` when the target name was different than expected but based on your recommendations this is no longer necessary. 

```
 # these tests require network access
  checkFlags = [
    "--skip=filehash::tests::remote_file_hash"
    "--skip=filehash::tests::get_filehash_unit_test"
  ];
```
Thanks for this. It's good to know there was a better way of skipping these particular tests.

Reply:
curious why you replaced the dependencies? `libiconv` for `zlib` Also, `linux-pam` for `pam`?

Reply:
(sorry for the late response, I did not see this question)

This was not a deliberate change, I just based the file posted above on the output of `nix-init` (and not your initial revision). On linux, pam is an alias for linux-pam, so this should not matter at least until further platforms are added. 

shavee doesn't use libiconv (I think?) and zlib is also added by [default-rust-crate-overrides.nix](https://github.com/NixOS/nixpkgs/blob/b7496f96eb8f53eabfe0f2d70c9116cc9d43cb79/pkgs/build-support/rust/default-crate-overrides.nix#L68), so it could be dropped.